src/client/v2/algod/block.ts

```diff
@@ -6,12 +6,10 @@ import HTTPClient from '../../client';
  * block gets the block info for the given round. this call may block
  */
 export default class Block extends JSONRequest {
```

Comment:
Why doesn't this one return `BlockResponse`?

Reply:
I'm not convinced it would be correct. The query uses msgp format which means the response will be the msgp-encoded raw block bytes.

Reply:
We use msgpack for a few other endpoints, like `PendingTransactions`, so I think that's something that can be handled.

What I see is a little different about this response is that its generated model contains a reference to `BlockHeader`, which is a handwritten definition in this SDK. I think this definition was typed based on the assumption that it would be JSON decoded, not msgpack decoded, so some of its types are not accurate -- e.g. msgpack binary data decodes as `Uint8Array` but there it's defined as `string`. Maybe this is what you were referring to?

I'd really like for us to return `BlockResponse` for this request as well, both for consistency and because it will make it much easier to make non-breaking improvements to the response type over time.

So I'd like to propose two options here:
1. Return `BlockResponse` and update our handwritten `BlockHeader` to have accurate types when decoded from msgpack.
2. Return `BlockResponse`, but change our handwritten `BlockHeader` to be a generic `Record<string, any>`. In the future we have the option of making this type more specific, which I don't think would necessarily be a breaking change. And even if we don't, users get some typing benefits (the top level response would be typed, with the two properties, `block` and `cert`).

Reply:
Tests pass for me when doing https://github.com/algorand/js-algorand-sdk/pull/776/commits/fef19974328ab7a31fdc47f24cd7585fb863025f 

Let me know what you think.